Framework/API/src/Algorithm.cpp

```diff
@@ -32,6 +32,9 @@
 
 #include <map>
 
+// Index property handling template definitions
+#include "MantidAPI/Algorithm.tcc"
```

Comment:
Yes, similar to what is done here: https://github.com/mantidproject/mantid/blob/master/Framework/DataObjects/src/WorkspaceProperty.cpp.